Signup: Update email validation message

StefanNieuwenhuis opened the pull request:
#### Changes proposed in this Pull Request
Changing the email address validation message to a more user-friendly alternative.

### Technical changes
* Copy is updated to _"An account with this email already exists. **Log in** or **reset your password**."_

#### Testing instructions
1. Check out this branch and `yarn && yarn start`
2. Apply D62261-code to your sandbox
3. Make sure that `public-api.wordpress.com` is proxied.
4. Go to `/start/user` create a new user with **an email address that's already registered**
5. Confirm that:
   * [ ] The validation message copy is updated to _"An account with this email already exists. **Log in** or **reset your password**."_
   * [ ] The "log in" link redirects to the existing 'login' page.
   * [ ] The "reset your password" link redirects to the existing 'reset your password' page.

#### Translation screenshot
![image](https://user-images.githubusercontent.com/17054134/121889510-9e618600-cd19-11eb-9393-86f46caeeeb9.png)

Fixes #53378

StefanNieuwenhuis committed:
Update email address validation msg

StefanNieuwenhuis commented:
https://www.loom.com/share/9ee1faf36e98487789b89bae796a50de

@ollierozdarz, what do you think of the copy when entering an email address that's already registered?

@alshakero suggested: "There is an existing account associated with this email address. If it's yours, login in now".

WDYT?

alshakero reviewed (commented):
Tested and this works as expected. But I think the message is unclear. It's not obvious that "welcome back" means you already have an account that is associated with your email address.

I'd do something like "There is an existing account associated with this email address. If it's yours, login in now".

StefanNieuwenhuis commented:
> Tested and this works as expected. But I think the message is unclear. It's not obvious that "welcome back" means you already have an account that is associated with your email address.
> 
> I'd do something like "There is an existing account associated with this email address. If it's yours, login in now".

Thanks for reviewing @alshakero and I'm happy that it works as expected! I agree with the ambiguous message and already asked Ollie about alternatives. I guess we have to await his response so I put this one on hold and added the `[Status] Needs Design` label :)

ollierozdarz commented:
👋 Hey @StefanNieuwenhuis, @alshakero! I think both ideas are great and have merit - friendly and informative. 

What about if we meet in the middle with the tone and say:

> An account with this email already exists. **Log in** or **reset your password**.

This is fairly consistent with some other sites:

- **Squarespace:** An account with this email already exists
- **Wix:** That email’s already in our system.
- **Dropbox:** This e-mail is already taken. Sign in or reset your password.

Any concerns?

StefanNieuwenhuis commented:
> 👋 Hey @StefanNieuwenhuis, @alshakero! I think both ideas are great and have merit - friendly and informative.
> 
> What about if we meet in the middle with the tone and say:
> 
> > An account with this email already exists. **Log in** or **reset your password**.
> 
> This is fairly consistent with some other larger sites:
> 
> * **Squarespace:** An account with this email already exists
> * **Wix:** That email’s already in our system.
> * **Dropbox:** This e-mail is already taken. Sign in or reset your password.
> 
> Any concerns?

I like this copy very much 👍 And I think it aligns nicely with the existing username validation message.

<img width="411" alt="Screenshot 2021-06-09 at 11 11 54" src="https://user-images.githubusercontent.com/12104589/121327375-ac776700-c913-11eb-9b14-e29db7241976.png">

@ollierozdarz Is it correct that:
1. The login link redirects to `/login`.
2. The reset link redirects to `/wp-login.php?action=lostpassword`.

alshakero commented:
> An account with this email already exists. Log in or reset your password.

I love it too!

ollierozdarz commented:
> * The login link redirects to `/login`.
> * The reset link redirects to `/wp-login.php?action=lostpassword`.

@StefanNieuwenhuis, yep, just to the existing 'login' and 'lost password' pages.

StefanNieuwenhuis commented:
> > * The login link redirects to `/login`.
> > * The reset link redirects to `/wp-login.php?action=lostpassword`.
> 
> @StefanNieuwenhuis, yep, just to the existing 'login' and 'lost password' pages.

Thanks for elaborating @ollierozdarz!

StefanNieuwenhuis committed:
Update validation msg for existing email

- Update copy
- Add login link
- Add pwd reset link
- Use lostPassword util to generate lost pwd link

alshakero reviewed (approved):
All works as advertised. LOVE THE NEW MESSAGE!

**Before**:
<img width="300" src="https://user-images.githubusercontent.com/17054134/121889477-9570b480-cd19-11eb-9e97-2faa82a96173.png">

**After**:
<img width="300" src="https://user-images.githubusercontent.com/17054134/121889510-9e618600-cd19-11eb-9393-86f46caeeeb9.png">

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6046096

Thank you @StefanNieuwenhuis for including a screenshot in the description! This is really helpful for our translators.

alshakero commented:
> could you please edit the description of this PR and add a screenshot for our translators?

Done.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6046096).
